stan/math/prim/arr/err/check_opencl.hpp

```diff
@@ -0,0 +1,304 @@
+#ifndef STAN_MATH_PRIM_ARR_ERR_CHECK_OPENCL_HPP_
+#define STAN_MATH_PRIM_ARR_ERR_CHECK_OPENCL_HPP_
+
+#include <CL/cl.hpp>
+#include <iostream>
+#include <string>
+
+
+/** @file stan/math/prim/arr/err/check_opencl.hpp
```

Comment:
This can be simplified with a throw function to avoid assignments and breaks.

I assume throwing on 0 was a mistake.

```
void throw_openCL(const std::string& function, const char* msg) {
  throw std::domain_error(function + ": The OpenCL application ended with the error: " + msg);
}
...

  switch(e.err()) {
    case 0:
      return;  // ??? should this throw instead ???
    case 1:
      throw_openCL("CL_DEVICE_NOT_FOUND");
...
```

Given that they're going to be Stan errors, we could also add some explanatory text.

Reply:
You are right, the throw on 0 was a mistake. The throw function was added as suggested. Thank you!